src/prepare.c

```diff
@@ -595,7 +615,7 @@ static inline void pthreads_prepare_constants(pthreads_object_t* thread) {
 						case IS_DOUBLE: Z_DVAL(constant.value)=Z_DVAL(zconstant->value); break;
 						case IS_STRING: {
 #if PHP_VERSION_ID >= 70300
-							Z_STR(constant.value)=Z_STR(zconstant->value);
+							Z_STR(constant.value)= zend_string_new(Z_STR(zconstant->value));
 #else
 							ZVAL_NEW_STR(&constant.value, zend_string_new(Z_STR(zconstant->value)));
 #endif
```

Comment:
This block probably does not need the if/else.

Reply:
yes, you're right